pkg/configurator/crd_methods.go

```diff
@@ -39,88 +39,88 @@ func (c *Client) GetConfigMap() ([]byte, error) {
 // where all existing traffic is allowed to flow as it is,
 // or it is in SMI Spec mode, in which only traffic between source/destinations
 // referenced in SMI policies is allowed.
-func (c *Client) IsPermissiveTrafficPolicyMode() bool {
-	return c.getConfigMap().PermissiveTrafficPolicyMode
+func (c *CRDClient) IsPermissiveTrafficPolicyMode() bool {
+	return c.getMeshConfig().PermissiveTrafficPolicyMode
 }
 
 // IsEgressEnabled determines whether egress is globally enabled in the mesh or not.
-func (c *Client) IsEgressEnabled() bool {
-	return c.getConfigMap().Egress
+func (c *CRDClient) IsEgressEnabled() bool {
+	return c.getMeshConfig().Egress
```

Comment:
As noted above, the conversion of `MeshConfig` -> `internal OsmConfig` is unnecessary. Can we just return the value from the MeshConfig struct instead?